### PR TITLE
#3348 Step three

### LINKED ISSUE #3348: New Sensor Flow/Page
## Is your feature request related to a problem? Please describe.

The new sensor flow and page

## Describe the solution you'd like

As per the marvel prototype

## Implementation

Quite a big page with a lot of steps. Going to break it down into the following tasks:

- [ ] `Stepper`
- [ ] Make `IconButton` have a `Rectangle` variant?
- [ ] Make step1 component
- [ ] Make step2 component
- [ ] Make step3 component
- [ ] Logic to send bluetooth commands + save a new sensor

## Describe alternatives you've considered

N/A

## Additional context

Resources

- https://marvelapp.com/prototype/55aea8f/screen/75648024
image

- https://drive.google.com/file/d/1wLg12JAQyrgB6D3F_X0O9LBZwJs8PcJM/view?usp=sharing
image

- https://drive.google.com/file/d/15SsY4YCmhpSgiihU1Q7FpZMnqCUkZcIp/view?usp=sharing
image

- https://drive.google.com/file/d/15SsY4YCmhpSgiihU1Q7FpZMnqCUkZcIp/view?usp=sharing
image

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3348 

## Change summary

- Adds step three
- Actually connecting and programming the sensor to come.

## Testing

Internal

### Related areas to think about

image

### CODE REVIEW

File: src/actions/NewSensorActions.js [hunk 1]
```diff
@@ -10,4 +14,25 @@ const updateConfig = (configType, configField, value) => ({
...
+
+const updateLogInterval = logInterval => ({
+  type: NEW_SENSOR_ACTIONS.UPDATE_LOG_INTERVAL,
+  payload: { logInterval },
+});
+
+const updateLoggingDelay = loggingDelay => ({
+  type: NEW_SENSOR_ACTIONS.UPDATE_LOGGING_DELAY,
+  payload: { loggingDelay },
+});
```
Comment (joshxg) [inline src/actions/NewSensorActions.js:29]: could have these as a single action with a field in the payload for which field in the new sensor to update. I kind of like the explicitness .. but then the action to update the configs is consistent with that.

I dunno

File: src/pages/NewSensor/NewSensorStepThree.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        </PaperSection>
+      </WithSpace>
+
+      <PaperSection height={130} contentContainerStyle={{ flex: 1 }}>
+        <EditorRow label={vaccineStrings.start_logging} Icon={<CalendarIcon color={DARKER_GREY} />}>
+          <DateEditor onPress={updateLoggingDelay} date={loggingDelay} />
+          <TimeEditor onPress={updateLoggingDelay} time={loggingDelay} />
+        </EditorRow>
+      </PaperSection>
+      <TextWithIcon left size="s" Icon={<HazardIcon color={LIGHT_GREY} />}>
```
Comment (joshxg) [inline src/pages/NewSensor/NewSensorStepThree.js:76]: Using this button row in a few places, was going to extract but didn't? (Could also utilize in dispensing pages.. possibly adding it into the 'wizard' component, but have to consider validation rules and communicating from the child (tab) to the parent (wizard) when to allow moving forward etc)

File: src/pages/NewSensor/NewSensorStepTwo.js [hunk 3]
```diff
@@ -8,29 +8,28 @@ import { BreachConfigRow } from './BreachConfigRow';
...
-      <View style={{ height: 280 }}>
-        <BreachConfigRow type="HOT_CONSECUTIVE" />
-        <BreachConfigRow type="COLD_CONSECUTIVE" />
-        <BreachConfigRow type="HOT_CUMULATIVE" />
-        <BreachConfigRow type="COLD_CUMULATIVE" />
-      </View>
+      <BreachConfigRow type="HOT_CONSECUTIVE" />
+      <BreachConfigRow type="COLD_CONSECUTIVE" />
+      <BreachConfigRow type="HOT_CUMULATIVE" />
+      <BreachConfigRow type="COLD_CUMULATIVE" />
```
Comment (joshxg) [inline src/pages/NewSensor/NewSensorStepTwo.js:21]: Added a flex:1 in `PaperSection` and could take away this view 🤷

File: src/pages/NewSensor/NewSensorStepThree.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  previousTab,
+}) => {
+  const withLoadingIndicator = useLoadingIndicator();
+
+  return (
+    <TabContainer>
+      <WithSpace space={10} horizontal={false} vertical>
+        <PaperSection height={200} headerText={vaccineStrings.new_sensor_step_three_title}>
+          <EditorRow label={vaccineStrings.sensor_name} Icon={<InfoIcon color={DARKER_GREY} />}>
+            <TextEditor size="large" value={name} onChangeText={updateName} />
```
Comment (mark-prins) [inline src/pages/NewSensor/NewSensorStepThree.js:54]: is it possible to left align the name text field - currently it's disconnected from the title and is more related to the code area

File: src/pages/NewSensor/NewSensorStepThree.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            <TextEditor label={vaccineStrings.sensor_code} value={code} onChangeText={updateCode} />
+          </EditorRow>
```
Comment (mark-prins) [inline src/pages/NewSensor/NewSensorStepThree.js:56]: are there plans for the (i) / info icon? To me, there should be info text when pressing on them and they are misleading without this happening.

File: src/pages/NewSensor/NewSensorStepThree.js (hunk 2 above)
Comment (mark-prins) [inline src/pages/NewSensor/NewSensorStepThree.js:76]: I find the light grey hazard icon weird. it looks like a disabled button to me

**Review by mark-prins — APPROVED** [review]
🎉  awesome 😄 

I made a few whiney comments. overall.. looks great. Go for it.

**Comment by joshxg** [comment]
Going to address your UI issues in a new PR where I will do some cleanup as the performance (switching tabs) is pretty suboptimal, also!

**Pull request merged**